src/server/pps/server/api_server.go

```diff
@@ -464,6 +467,248 @@ func (a *apiServer) RestartDatum(ctx context.Context, request *pps.RestartDatumR
 	return &types.Empty{}, nil
 }
 
+func (a *apiServer) ListDatum(ctx context.Context, request *pps.ListDatumRequest) (response *pps.DatumInfos, retErr error) {
+	func() { a.Log(request, nil, nil, 0) }()
+	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
+	jobInfo, err := a.InspectJob(ctx, &pps.InspectJobRequest{
+		Job: &pps.Job{
+			ID: request.Job.ID,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	pfsClient, err := a.getPFSClient()
+	if err != nil {
+		return nil, err
+	}
+	// Check 'stats' branch exists
+	branches, err := pfsClient.ListBranch(ctx, &pfs.ListBranchRequest{jobInfo.OutputCommit.Repo})
+	if err != nil {
+		return nil, err
+	}
+	var statsBranch *pfs.BranchInfo
+	for _, branch := range branches.BranchInfo {
+		if branch.Name == "stats" {
+			statsBranch = branch
+		}
+	}
+	if statsBranch == nil {
+		return nil, fmt.Errorf("stats branch not found on %v", jobInfo.OutputCommit.Repo.Name)
+	}
+	// List the files under /jobID to get all the datums
+	file := &pfs.File{
+		Commit: statsBranch.Head,
```

Comment:
I don't think you want to use `statsBranch.Head` here since that means that the result of `ListDatum` will change as new jobs run. Instead just use `jobInfo.StatsCommit` and if that's `nil` you know there's no stats output. This should also allow you to throw out several lines above.